src/libbson/src/bson/bson-atomic.h

```diff
@@ -32,6 +32,8 @@
 
 BSON_BEGIN_DECLS
 
+MC_DISABLE_COVERED_SWITCH_DEFAULT_BEGIN
```

Comment:
Confirming: `-Wcovered-switch-default` is suppressed, rather than `default` removed, because it contradicts the warnings from `-Wswitch-default`, correct?

Reply:
Yes, removing `default` will then trigger the `-Wswitch-default` warning (even though all of the switch statements deal with enums that are fully covered by the cases). I tried removing the `default` case and instead `ASSERT`ing that the switched value is one of the enums, but the compiler didn't like that either. 

I think the only solution would be to remove the final `case` of each `switch` and replace it with the `default`. I don't love this idea though because it makes the code less readable and could cause issues later on if another value is added to the enum (new enum value would be covered by `default` instead of throwing an error, putting the ownership on the developer to remember they have to add a case to the switch statement).

Reply:
Interesting. Thank you for the explanation. I agree with the current disabling the warning. Resolving.